lib/commands/query.js

```diff
@@ -17,6 +17,7 @@ class QuerySelectorItem {
     this.resolved = node.target.resolved
     this.isLink = node.target.isLink
     this.isWorkspace = node.target.isWorkspace
+    // XXX should we have dev, optional, devOptional?
```

Comment:
dev and optional exist, so yes, devOptional would be great but i don't think it's a thing :-)

Reply:
It is an attribute on arborist nodes.

Reply:
interesting. where does it come from, since package.json doesn't support devOptionalDependencies? or is it "optional deps of a dev dep"

Reply:
`devOptional is more internal to arborist, only adding `dev` `optional` `peer` from the node.